### PR TITLE
Share config between util and functional tests

### PR DESCRIPTION
**Pull request by jnewbery**
The functional tests and util tests both require a config file that is
generated by ./configure. This PR merges those two config
files into a single configuration file that can be shared by both tests.

This PR also merges bctest.py and bitcoin-util-test.py into a single module.
bctest.py is only used as an import by bitcoin-util-test.py. There's no
value in keeping it as a separate module, so this PR merges them into a
single module to keep building and packaging simpler.

Note - this bumps the required python version for bitcoin-util-test.py to python 3.

### CODE REVIEW

**Comment by jnewbery** [comment]
I have tested this locally, both for an in-tree and out-of-tree build.

**Review by jimmysong — COMMENTED** [review]
utACK 3fe56f2860a2226b1295ffaded5817098ac5898d

Figuring out exactly how to test this

**Comment by jimmysong** [comment]
Tested ACK 3fe56f2860a2226b1295ffaded5817098ac5898d

**Comment by fanquake** [comment]
Do we need to mention the Python 3 requirement somewhere?

**Comment by laanwj** [comment]
> Do we need to mention the Python 3 requirement somewhere?

So the difference is that 'make check' now requires Python 3?

**Comment by jnewbery** [comment]
> So the difference is that 'make check' now requires Python 3?

Correct. That would also be the case for #10044 (see discussion here: https://botbot.me/freenode/bitcoin-core-dev/2017-03-16/?msg=82548466&page=5)

**Comment by sipa** [comment]
Concept ACK

File: test/util/bctest.py [hunk 1]
```diff
@@ -1,139 +0,0 @@
...
-
-def bctester(testDir, input_basename, buildenv):
-    """ Loads and parses the input file, runs all tests and reports results"""
-    input_filename = testDir + "/" + input_basename
-    raw_data = open(input_filename).read()
-    input_data = json.loads(raw_data)
-
-    failed_testcases = []
-
-    for testObj in input_data:
```
Comment (TheBlueMatt) [inline test/util/bctest.py]: This block didnt make the move.

**Review by TheBlueMatt — COMMENTED** [review]
Otherwise looks good.

**Comment by jnewbery** [comment]
Thanks @TheBlueMatt . Block restored in fixup commit. I'll squash when this is ready for merge.

**Comment by MarcoFalke** [comment]
Please squash the two fixup commits.

**Comment by jnewbery** [comment]
fixup commits squashed

**Comment by MarcoFalke** [comment]
utACK 8ad5bde

**Pull request merged**